### ISSUE TITLE
Random failure in Test_SphericalCompression

### ISSUE THREAD

**Issue by kidder** [sxs-collaboration/spectre#2994]
# Bug reports:

### Expected behavior:

Test passes

### Current behavior:

see CI Failure at
https://github.com/sxs-collaboration/spectre/pull/2990/checks?check_run_id=2126577086
```
Unit.Domain.CoordinateMaps.SphericalCompression
-------------------------------------------------------------------------------
/__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:468
...............................................................................

/__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:234: FAILED:
  CHECK_FALSE( map.inverse(point(target_radius), time, functions_of_time).has_value() )
with expansion:
  !true
with message:
  Seed is: 3858650112 from /__w/spectre/spectre/tests/Unit/Domain/
  CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:470
```

**Comment by nilsdeppe** [comment]
And also:
```
2021-03-17T17:51:29.6564929Z  227/1208 Test  #286: Unit.Domain.CoordinateMaps.SphericalCompression .............................***Failed    0.20 sec
2021-03-17T17:51:29.6566379Z Charm++> No provisioning arguments specified. Running with a single PE.
2021-03-17T17:51:29.6569544Z          Use +auto-provision to fully subscribe resources or +p1 to silence this message.
2021-03-17T17:51:29.6570382Z Charm++: standalone mode (not using charmrun)
2021-03-17T17:51:29.6571285Z Charm++> Running in Multicore mode: 1 threads (PEs)
2021-03-17T17:51:29.6572160Z Converse/Charm++ Commit ID: v6.10.2-0-g7bf00fa
2021-03-17T17:51:29.6573752Z Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
2021-03-17T17:51:29.6575014Z CharmLB> Load balancer assumes all CPUs are same.
2021-03-17T17:51:29.6575903Z Charm++> Running on 1 hosts (1 sockets x 2 cores x 1 PUs = 2-way SMP)
2021-03-17T17:51:29.6576622Z Charm++> cpu topology info is gathered in 0.000 seconds.
2021-03-17T17:51:29.6577237Z SpECTRE Build Information:
2021-03-17T17:51:29.6577735Z Version:                      2021.03.09
2021-03-17T17:51:29.6578222Z Compiled on host:             ca179ea369b4
2021-03-17T17:51:29.6578795Z Compiled in directory:        /work/build
2021-03-17T17:51:29.6579403Z Source directory is:          /__w/spectre/spectre
2021-03-17T17:51:29.6580002Z Compiled on git branch:       HEAD
2021-03-17T17:51:29.6580556Z Compiled on git revision:     0ba0ed8
2021-03-17T17:51:29.6581077Z Linked on:                    Wed Mar 17 17:51:02 2021
2021-03-17T17:51:29.6582191Z Filters: "Unit.Domain.CoordinateMaps.SphericalCompression"
2021-03-17T17:51:29.6583110Z 
2021-03-17T17:51:29.6583494Z ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-03-17T17:51:29.6584882Z "Release.Unit.Domain.CoordinateMaps.SphericalCompression" is a Catch v2.11.1 host application.
2021-03-17T17:51:29.6586440Z Run with -? for options
2021-03-17T17:51:29.6586951Z 
2021-03-17T17:51:29.6587692Z -------------------------------------------------------------------------------
2021-03-17T17:51:29.6588876Z Unit.Domain.CoordinateMaps.SphericalCompression
2021-03-17T17:51:29.6590230Z -------------------------------------------------------------------------------
2021-03-17T17:51:29.6591365Z /__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:469
2021-03-17T17:51:29.6592328Z ...............................................................................
2021-03-17T17:51:29.6592730Z 
2021-03-17T17:51:29.6593660Z /__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:235: FAILED:
2021-03-17T17:51:29.6594890Z   CHECK_FALSE( map.inverse(point(target_radius), time, functions_of_time).has_value() )
2021-03-17T17:51:29.6595596Z with expansion:
2021-03-17T17:51:29.6596005Z   !true
2021-03-17T17:51:29.6596376Z with message:
2021-03-17T17:51:29.6596952Z   Seed is: 3804391669 from /__w/spectre/spectre/tests/Unit/Domain/
2021-03-17T17:51:29.6597898Z   CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:470
2021-03-17T17:51:29.6598560Z 
2021-03-17T17:51:29.6599530Z 0.015 s: Unit.Domain.CoordinateMaps.SphericalCompression
2021-03-17T17:51:29.6600563Z ===============================================================================
2021-03-17T17:51:29.6601050Z test cases:    1 |    0 passed | 1 failed
2021-03-17T17:51:29.6601593Z assertions: 1457 | 1456 passed | 1 failed
2021-03-17T17:51:29.6601944Z 
2021-03-17T17:51:29.6602327Z A catch test has failed.
2021-03-17T17:51:29.6602844Z [Partition 0][Node 0] End of program
```

**Comment by nilsdeppe** [comment]
And another:
```
2021-03-17T14:47:18.0724537Z /__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:235: FAILED:
2021-03-17T14:47:18.0725656Z   CHECK_FALSE( map.inverse(point(target_radius), time, functions_of_time).has_value() )
2021-03-17T14:47:18.0726313Z with expansion:
2021-03-17T14:47:18.0726672Z   !true
2021-03-17T14:47:18.0727022Z with message:
2021-03-17T14:47:18.0727533Z   Seed is: 4044964464 from /__w/spectre/spectre/tests/Unit/Domain/
2021-03-17T14:47:18.0728407Z   CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:470
```

**Comment by prayush** [comment]
Again (on #2528 ):

```
/__w/spectre/spectre/tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:235: FAILED:
  CHECK_FALSE( map.inverse(point(target_radius), time, functions_of_time).has_value() )
with expansion:
  !true
with message:
  Seed is: 385472744 from /__w/spectre/spectre/tests/Unit/Domain/
  CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp:470
```

**Comment by nilsdeppe** [comment]
Should be fixed with #2996
